bin/spack.ps1

```diff
@@ -39,6 +39,15 @@ function Read-SpackArgs {
     return $SpackCMD_params, $SpackSubCommand, $SpackSubCommandArgs
 }
 
+function Set-SpackEnv {
+    foreach($envop in $args[0]){
+        $envop = $envop.Trim("`$")
```

Comment:
The point of an option like `--pswh` in `spack env deactivate` is to output a string that, if we typed it into Powershell` would achieve the effect we want. You can see similar logic for `spack env deactivate --sh` in `setup-env.sh` (we `eval` it). In general, running arbitrary strings is subject to injection, but we generate them ourselves, so that's not an issue. Unless you think there are distinct risks, I'd rather not sidestep the (loose) interface we created for this.

The article you linked suggests `&` which may help us from messing up too much if we generate the wrong commands for ourselves (can we `& "$output"`?).

> Note: Set-SpackEnv is being run after each command intercepted by the script, not just deactivate. I assume you were just using deactivate as an example, 

Yes, I think the concept generalizes to activate/load as well.

Reply:
> running arbitrary strings is subject to injection, but we generate them ourselves, so that's not an issue

My concern was less with the security perspective and more with "corner case env variables" that may give `Invoke-Expression` a hard time w.r.t. to being empty, quoted, spaced, etc. I think ensuring we're quoting paths when we set env variables (a task out side the scope of this PR I'm working on) would mitigate the vast majority of my concerns there, and that can be a quick follow up effort. 
I tested a few alternative implementations, and they either had the same level of fragility as `Invoke-Expression` or were more circuitous than the original approach. `Invoke-Expresion` it is.